packages/ses-ava/src/ses-ava-test.js

```diff
@@ -110,8 +110,10 @@ const wrapTester = (testerFunc, logger = defaultLogger) => {
  * @returns {TesterInterface}
  */
 const wrapTest = (avaTest, logger = defaultLogger) => {
-  /** @type {TesterInterface} */
-  const testerWrapper = wrapTester(avaTest, logger);
+  const testerWrapper = /** @type {TesterInterface} */ (wrapTester(
```

Comment:
It’s not avoidable in this case. The only reason the former pattern was allowed was because this file wasn’t properly covered by the type checker.

Reply:
It's always avoidable by using `@ts-ignore` or `@ts-expect-error` instead, right?

Aside from that, if it is unavoidable, how come the before code (code in the left pane) doesn't need one? It locally looks otherwise identical.

Reply:
Because before this PR, the result of `wrapTester` was likely `any`, so assigning that to something of a stricter type was ok. Now I'm not sure what the return type of `wrapTester` is (it's possible it could be better typed to avoid this issue), but it seem to be incompatible with `TesterInterface`.

adding a `ts-ignore` would basically make `testerWrapper` typed as `any` so I consider it inferior. It also may silence multiple errors.

Reply:
A `@ts-ignore` comment does suppress the error on the line, and I believe that makes `testerWrapper`’s type `any`, which disables any internal coherence checking from that point to the end of the function. If we did not also have an explicit `@returns` pragma, the `any` type would propagate to the return type.

I believe the difference between before and after here is that in the before case, `testerWrapper` is being labeled as a `TesterInterface` but the return type of `wrapTester` is `TestFunc`, which is not compatible. In the after case, we are forcing the result of `wrapTester` from `TestFunc` to `TesterInterface`, which makes the const `testerWrapper` a `TesterInterface` by inference.